src/System.Text.Json/tests/Serialization/Stream.WriteTests.cs

```diff
@@ -60,6 +59,29 @@ public static async Task RoundTripAsync()
             }
         }
 
+        [Fact]
+        public static async Task WriteLargeJson()
+        {
+            // Generating tailored json
+            int i = 0;
+            StringBuilder json = new StringBuilder();
+            json.Append("{");
+            while (true)
+            {
+                if (json.Length >= 14757)
+                {
+                    break;
+                }
+                json.AppendFormat(@"""Key_{0}"":""{0}"",", i);
+                i++;
+            }
+            json.Remove(json.Length - 1, 1).Append("}");
+
+            JsonElement root = JsonSerializer.Parse<JsonElement>(json.ToString());
+            var ms = new MemoryStream();
+            await JsonSerializer.WriteAsync(ms, root, root.GetType());
```

Comment:
Is this only an issue for the Async API that takes a stream?

Can you also add similar tests but for the other synchronous overloads (ToString, ToUtf8Bytes)?

Reply:
Yea, I tried both and only the Stream one failed.